pkg/services/services.go

```diff
@@ -0,0 +1,239 @@
+package services
+
+import (
+	"errors"
+	"fmt"
+	"sync"
+	"sync/atomic"
+
+	pkgerrors "github.com/pkg/errors"
+)
+
+// defaultErrorBufferCap is the default cap on the errors an error buffer can store at any time
+const defaultErrorBufferCap = 50
+
+type errNotStarted struct {
+	state State
+}
+
+func (e *errNotStarted) Error() string {
+	return fmt.Sprintf("service is %q, not started", e.state)
+}
+
+var (
+	ErrAlreadyStopped      = errors.New("already stopped")
+	ErrCannotStopUnstarted = errors.New("cannot stop unstarted service")
+)
+
+// StateMachine contains a State integer
+type StateMachine struct {
```

Comment:
This is only type that isn't a straight copy. It is functionally identical to the latest core version, but with some things renamed and/or unexported.